impl/graphsync.go

```diff
@@ -304,6 +307,17 @@ func New(parent context.Context, network gsnet.GraphSyncNetwork,
 
 // Request initiates a new GraphSync request to the given peer using the given selector spec.
 func (gs *GraphSync) Request(ctx context.Context, p peer.ID, root ipld.Link, selector ipld.Node, extensions ...graphsync.ExtensionData) (<-chan graphsync.ResponseProgress, <-chan error) {
+	var extNames []string
+	for _, ext := range extensions {
+		extNames = append(extNames, string(ext.Name))
+	}
+	var span trace.Span
+	ctx, span = otel.Tracer("graphsync").Start(ctx, "request", trace.WithAttributes(
+		attribute.String("peerID", p.Pretty()),
+		attribute.String("root", root.String()),
+		attribute.StringSlice("extensions", extNames),
+	))
+	defer span.End()
```

Comment:
so this is the issue I think with spans here -- your span.End() runs as soon as the Request method on request manager finishes, which is NOT the same as when the request actually ends. I think you had better do this in RequestManager.Request, and pass the span into the go routing that collects responses, and only span.End at the end of that.